extensions/big-data-cluster/src/extension.ts

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import { ControllerTreeDataProvider } from './bigDataCluster/tree/controllerTreeDataProvider';
+import { IconPath } from './bigDataCluster/constants';
+import { TreeNode } from './bigDataCluster/tree/treeNode';
+import { AddControllerDialogModel, AddControllerDialog } from './bigDataCluster/dialog/addControllerDialog';
+import { ControllerNode } from './bigDataCluster/tree/controllerTreeNode';
+
+const localize = nls.loadMessageBundle();
+
+export function activate(extensionContext: vscode.ExtensionContext) {
+	IconPath.setExtensionContext(extensionContext);
+	let treeDataProvider = new ControllerTreeDataProvider();
+
+	registerTreeDataProvider(treeDataProvider);
+	registerCommands(treeDataProvider);
+}
+
+export function deactivate() {
+}
+
+function registerTreeDataProvider(treeDataProvider: ControllerTreeDataProvider): void {
+	vscode.window.registerTreeDataProvider('sqlBigDataCluster', treeDataProvider);
+}
+
+function registerCommands(treeDataProvider: ControllerTreeDataProvider): void {
+	vscode.commands.registerCommand('bigDataClusters.command.addController', (node?: TreeNode) => {
+		addBdcControllerCommand(treeDataProvider, node);
+	});
+
+	vscode.commands.registerCommand('bigDataClusters.command.deleteController', (node: TreeNode) => {
+		deleteBdcControllerCommand(treeDataProvider, node);
+	});
+
+	vscode.commands.registerCommand('bigDataClusters.command.refreshController', (node: TreeNode) => {
+		if (!node) {
+			return;
+		}
+		treeDataProvider.notifyNodeChanged(node);
+	});
+}
+
+function addBdcControllerCommand(treeDataProvider: ControllerTreeDataProvider, node?: TreeNode): void {
```

Comment:
minor: we can just name this `addBdcController`

Reply:
Same for "delete" function below

Reply:
fixed.